Azure/sonic-buildimage
# secureboot: Enable signing SONiC kernel

Build hook to enable signing SONiC Linux kernel using standard third party sbsign tool

   How to enable signing hook?
   Pass following parameter to rules/config.user
   Ex:
   SONIC_ENABLE_SECUREBOOT_SIGNATURE := y
   SIGNING_KEY := /path/to/key/private.key
   SIGNING_CERT := /path/to/public/public.cert


Signed-off-by: Sachin Naik <sachnaik@cisco.com>



#### Why I did it
To sign SONiC kernel image and allow secure boot based system to verify SONiC image before loading into the system.

#### How I did it
   Pass following parameter to rules/config.user
   Ex:
   SONIC_ENABLE_SECUREBOOT_SIGNATURE := y
   SIGNING_KEY := /path/to/key/private.key
   SIGNING_CERT := /path/to/public/public.cert

#### How to verify it

Secure boot enabled system enrolled with right public key of the, image in the platform UEFI database will able to verify image before load. 

Alternatively one can verify with offline sbsign tool as below.

export SBSIGN_KEY=/abc/bcd/xyz/
sbverify --cert $SBSIGN_KEY/public_cert.cert  fsroot-platform-XYZ/boot/vmlinuz-5.10.0-8-2-amd64 mage

O/P:
Signature verification OK

#### Which release branch to backport (provide reason below if selected)


202012 branch need to be backported so that it can produce signed SONiC image to boot on secure boot enabled hardware.


#### Description for the changelog

Generic build hook to sign SONiC kernel image. It provides mechanism to configure SONiC build system to sign the image. The user can enable signing via make configure and provide a location to signing keys. It uses standard sbsign infra to sign Linux kernel. 

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
